apisix/admin/standalone.lua

```diff
@@ -158,6 +157,120 @@ local function check_conf(checker, schema, item, typ)
+                local valid, err = check_conf(item_checker, item_schema, item_temp, key)
+                if not valid then
+                    local err_prefix = "invalid " .. key .. " at index " .. (index - 1) .. ", err: "
+                    local err_msg = type(err) == "table" and err.error_msg or err
+                    local error_msg = err_prefix .. err_msg
+
+                    if not collect_all_errors then
+                        return false, error_msg
+                    end
+                    validation_results.valid = false
+                    table_insert(validation_results.errors, {
+                        resource_type = key,
+                        index = index - 1,
+                        error = error_msg
+                    })
+                end
+
+                -- check for duplicate IDs
+                local duplicated, dup_err = check_duplicate(item, key, id_set)
+                if duplicated then
+                    if not collect_all_errors then
+                        return false, dup_err
+                    end
+                    validation_results.valid = false
+                    table_insert(validation_results.errors, {
+                        resource_type = key,
+                        index = index - 1,
+                        error = dup_err
+                    })
+                end
+            end
+        end
+    end
+
+    if collect_all_errors then
+        return validation_results.valid, validation_results
+    else
+        return validation_results.valid, nil
+    end
+end
+
+local function validate(ctx)
+    local content_type = core.request.header(nil, "content-type") or "application/json"
+    local req_body, err = core.request.get_body()
+    if err then
+        return core.response.exit(400, {error_msg = "invalid request body: " .. err})
+    end
+
+    if not req_body or #req_body <= 0 then
+        return core.response.exit(400, {error_msg = "invalid request body: empty request body"})
+    end
+
+    local data
+    if core.string.has_prefix(content_type, "application/yaml") then
+        local ok, result = pcall(yaml.load, req_body, { all = false })
+        if not ok or type(result) ~= "table" then
+            err = "invalid yaml request body"
+        else
+            data = result
+        end
+    else
+        data, err = core.json.decode(req_body)
+    end
+
+    if err then
+        core.log.error("invalid request body: ", req_body, " err: ", err)
+        return core.response.exit(400, {error_msg = "invalid request body: " .. err})
+    end
+
+    local valid, validation_results = validate_configuration(data, true)
+    if not valid then
+        return core.response.exit(400, {
+            error_msg = "Configuration validation failed",
+            valid = false,
+            errors = validation_results.errors
+        })
+    end
+
+    return core.response.exit(200, {
+        message = "Configuration is valid",
+        valid = true
+    })
```

Comment:
When the check passes, neither the `message` nor the `valid` field carries any meaning; an HTTP 200 response is sufficient to imply this outcome. In practice, neither of these return fields will be utilised anywhere. **Remove them.**

When the check fails, the `valid` field similarly holds no significance. An HTTP 400 response can indicate this error. **Remove it.** The `error_msg` should maintain a consistent style, **using lowercase initial letters.**